### PR TITLE
:mortar_board: Topic ESAP ram --- Start explaining the module 

### PR DESCRIPTION
**Pull request by jameshughes89**
### What

Start explaining the RAM module by
1. reviewing RAM
2. Explaining how we will build ours

### Testing

:+1: 

### Additional Notes

The content was going to be more involved, but I simplified it. It _was_ going to include more details on why it's only using 4 bits for addressing and showing how the RAM component was configured in Digital. However, this seemed unnecessary in the end.

### CODE REVIEW

File: site/topics/memory/esap-ram.rst [hunk 1]
```diff
@@ -16,6 +16,75 @@ Address Register and RAM
 RAM Module
 ==========
 
+* As previously discussed, memory stores data in individually indexable memory locations called addresses
+* The number of unique memory addresses is called the address space
+* Each location stores some amount of data, called the addressability
```
Comment (twentylemon) [inline site/topics/memory/esap-ram.rst:21]: I've never heard the term _addressability_ used in this context. Word.

File: site/topics/memory/esap-ram.rst [hunk 2]
```diff
@@ -16,6 +16,75 @@ Address Register and RAM
...
+
+
+* This means the system can store up to 16 bytes of data in RAM
+
+
+.. note::
+
+    Until the late 2000s/early 2010s, a typical computer used a 32 bit processor. One of the limitations of such a
+    system is the address space --- 32 bits could index :math:`2^{32}`, or :math:`4,294,967,296` unique memory
+    addresses. This may feel like a lot, but to put this into perspective, that's only :math:`4GB`.
```
Comment (twentylemon) [inline site/topics/memory/esap-ram.rst]: Not actually true for pentium chips! They actually had a 36 bit address space, for some reason. People invented some wild shit which is now completely useless.

**Review by twentylemon — APPROVED** [review]

File: site/topics/memory/esap-ram.rst (hunk 2 above)
Comment (jameshughes89) [reply site/topics/memory/esap-ram.rst]: Do you mean, it was not a 32 bit processor, but was some processor that had a 32 bit address space?

Comment (twentylemon) [reply site/topics/memory/esap-ram.rst]: Yurp. All the the pentiums >=4 had it iirc.

File: site/topics/memory/esap-ram.rst (hunk 1 above)
Comment (jameshughes89) [reply site/topics/memory/esap-ram.rst:21]: It does feel odd to use it _this way_. 

I feel like the word is used more to describe the resolution (for lack of a better term) of memory's ability to address data. Like, a measure of how fine we can get data out of RAM. In a typical comp, it's byte addressable, meaning we can pin point individual bytes, which is cool. 

However, the flip side is, that's how much data is stored in each address' location. For the purposes here, it's effectively talking about the same thing, _but_ the wording does feel odd.

**Pull request merged**